Raspberry Pi 4 board support

rvs opened the pull request:
As usual, the way to review this is commit by commit

eriknordmark commented on pkg/fw/rpi/brcm/brcmfmac43340-sdio.meegopad-t08.txt:
```diff
@@ -0,0 +1,98 @@
+# SPDX-License-Identifier: GPL-2.0+
```
Is this the first occurance of GPL source in EVE? (We have plenty of packages we pull in from Alpine and Xen which are GPL, but I haven't seen source files.)

rvs replied:
Got catch @eriknordmark -- but we already have quite a bit of pre-existing GPL code (in GRUB and a few other places). I think the trick for now is to make sure that it is extremely clearly demarkated from everything else (which I hope I done with those README files).

One additional note on these particular blobs is that these are going away once the upstream linux-firmware Alpine package catches up (typically in a month or so).

The other blobs (the RPi bootloader ones) will stay -- but we can segregate them even further over time.

eriknordmark commented on conf/grub.cfg.kvm:
```diff
@@ -0,0 +1,5 @@
+# You can put your GRUB overrides here
+set_global load_hv_cmd echo
+set_global load_dom0_cmd linux
+set_global dom0_console console=tty0
+set_global load_initrd_cmd initrd
```
One would need to rename this file to /config/grub.cfg for it to come into effect, right?

Would make sense to put that in eve/docs somewhere.

rvs replied:
Absolutely @eriknordmark -- let me update the README

rvs commented:
> I see the Makefile rule for live rpi image, but how do I build a rootfs?

rootfs is built the normal way -- but you'll have to tweak the pointer to the kernel package. Now that you suggested a README update -- let me document those few steps.

Btw, in general, this is just an issue with the kernel version -- RPi requires a really recent (like of two days ago recent -- I'm not kidding) upstream Linux kernel. In a way, this is a good way for us to test EVE with the kernel we're likely going to move to in 2020 (since 4.x series are getting pretty old at this point). But I still haven't figured out a good way to manage multiple kernel packages.

One idea I had is to may be have a pkg/kernel-next package to track the future kernels.

> Does that always use the 5.6 kernel? And show we use it for other ARM?

Yes it does -- and, it is with extreme sadness, that I have to report that HiKey still not compatible with 5.x series. I will ping Linaro folks about that -- but based on previous interactions I think it'll be a pretty involved exersize that will also require HiKey firmware updates, etc. etc.

Now, the only reason HiKey is still interesting to us is that there's some issue with Xen in my build on RPi that I'm debugging -- so may be I'll debug it sooner that we can enabled HiKey on 5.x kernels -- and then we can just ditch HiKey alltogether.

> Any noticable size impact on the rootfs?

There's some (b/c of double kernel modules) but it is manageable +18Mb or so.

> Finally, let me know if and when you want me to take this on a spin on a hikey.

Actually ;-) what I would really appreciate you testing on HiKey is #826 

Lets make sure the basic KVM works there and then we can test it in parallel with RPi.

So... please review #826 -- lets merge it in, test it on HiKey and then come back to RPi

rvs commented:
Hey @eriknordmark -- I've fixed a lot of things, simplified quite a bit of others and provided a README https://github.com/rvs/eve/tree/rpi#how-to-use-on-a-raspberry-pi-4-arm-board

Please let me know if you feel like this could be merged

deitch commented on tools/parse-pkgs.sh:
```diff
@@ -7,6 +7,9 @@
 PATH="$EVE/build-tools/bin:$PATH"
 
+# Version of the Linux kernel that we build for early testing
+NEW_KERNEL_VERSION=5.6.1
```
Curious to know why you moved this into here from the Makefile. `parse-pkgs` usually uses logic to determine pkg versions. Hard-coding it in here is going to be hard to track down later.

rvs replied:
This is basically to make NEW KERNEL a fake package. In fact, doing it convinced me that we should probably make it a real pkg/new-kernel

What do you think @deitch ?

deitch commented on Makefile:
```diff
@@ -240,6 +245,10 @@ $(LIVE_IMG).qcow2: $(LIVE_IMG).raw | $(DIST)
 	qemu-img convert -c -f raw -O qcow2 $< $@
 	rm $<
 
+$(LIVE_IMG).rpi: $(BOOT_PART) $(EFI_PART) $(ROOTFS_IMG) $(CONFIG_IMG) | $(INSTALLER)
```
Does this change the disk layout between RPi and other systems? Does it have a fundamentally different layout, with a different number of partitions? If so, we should document it and why it is necessary.

rvs replied:
It doesn't change the disk layout, but it does change how the GRUB trampoline hand-off works (see the comment right above this in the later PR @deitch). As for documenting it -- let me first ponder on how this could be generalized to other systems.

In fact, let me ponder if we shouldn't be transitioning to readconfig style of a trampoline across the board.

In the meantime -- I think it is fine for this "twist" to stay like it is.

rvs commented:
> Finally had a chance to review. There is a lot in here. I found the README a bit confusing. In the end, it is just "run these steps slightly differently than normal", but the preceding paragraph threw me off. If I were coming from the outside, I would be somewhat discouraged - it sounds like, "this has to go through lots of hoops to work, and it all is boiled into this one paragraph, but if you really want to go through the pain, read these next steps." And then the next steps aren't so bad!

Well, the steps are not bad @deitch  but:
   1. there's quite a bit happening behind the scenes in Makefile logic (the kind of things that don't happen for other images)
   2. the resulting image is limited compared to the same (default) image for HiKey

It is #2 that I think is fair for us to warn our users about (note that this may change once I enabled Xen on RPi and then I'd be more than happy to follow your advice).

> I don't have any issues with the content of the PR, but @deitch has some comments to look at.
My remaining concern is that since we had to back out the code to enable cross compile, the instructions will not currently work. 

Actually @eriknordmark -- the instructions are working perfectly without any kind of cross compile. Now, you won't be able to build you pillar code if you want to experiment -- but that's a much more involved operation anyway (and is only interesting to EVE developers not EVE users).

rvs commented:
Hey @deitch -- I've update the README somewhat -- please take a look
